relay_messages.go

```diff
@@ -132,24 +146,40 @@ func newLazyCallReq(f *Frame) (*lazyCallReq, error) {
 			cr.key = val
 		}
 	}
+	cr.headerEndOffset = rbufOffset()
 
 	// csumtype:1 (csum:4){0,1} arg1~2 arg2~2 arg3~2
-	checkSumType := ChecksumType(rbuf.ReadSingleByte())
-	rbuf.ReadBytes(checkSumType.ChecksumSize())
+	cr.checksumType = ChecksumType(rbuf.ReadSingleByte())
+	rbuf.ReadBytes(cr.checksumType.ChecksumSize())
 
 	// arg1~2
 	arg1Len := int(rbuf.ReadUint16())
 	cr.method = rbuf.ReadBytes(arg1Len)
 
 	// arg2~2
 	arg2Len := int(rbuf.ReadUint16())
-	cr.arg2StartOffset = int(headerStartIndex) + rbuf.BytesRead()
-	// TODO(echung): deprecate the arg2 offsets and store just the arg2 slice instead
-	// Read past arg2
```

Comment:
eventually decided against this as the effort doesn't quite justify the return - a lot of tests depend on knowing the arg2 offsets, and switching to a slice may actually slightly hurt performance, so giving up